Assembler.Base/EndFrameHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public class EndFrameHandler<TFrame, TMessage> : BaseFrameHandler<TFrame, TMessage>
+        where TFrame : BaseFrame
+        where TMessage : BaseMessageInAssembly
+    {
+        private readonly bool _isToReleaseSingleEndFrame;
+        private readonly ILogger _logger;
+
+        public EndFrameHandler(ITimeBasedCache<TMessage> cache,
+            IFactory<TFrame, string> identifierFactory, IMessageEnricher<TFrame, TMessage> enricher,
+            ICreator<TMessage> assembledMessageCreator, bool isToReleaseOnlyEndFrame,
+            ILoggerFactory loggerFactory) : base(cache, identifierFactory, enricher, assembledMessageCreator)
+        {
+            _isToReleaseSingleEndFrame = isToReleaseOnlyEndFrame;
+            _logger = loggerFactory.GetLogger(this);
+        }
+
+        public override void Handle(TFrame frame)
+        {
+            string identifier;
+
+            try
+            {
+                identifier = GetIdentifier(frame);
+            }
+            catch (Exception e)
+            {
+                _logger.Error($"There was an error while getting an identifier out of the frame [{frame.Guid}], " +
+                              $"it won't be used it the assembling process \n {e}");
+                return;
+            }
+
+            if (TryGetOrCreateMessageInAssembly(identifier, out var message))
+            {
+                MessageEnricher.Enrich(frame, message);
+
+                _logger.Debug(
+                    $"Enriched [{message.Guid}] with the frame [{frame.Guid}].");
+
+                message.ReleaseReason = ReleaseReason.EndReceived;
+                ReleaseMessage(message);
+            }
+            else
+            {
+                _logger.Debug(
```

Comment:
Those lines will be called only when _isToReleaseSingleEndFrame set to false, otherwise that code will never be reached...

Are you OK with that?

Reply:
Yes, don't you?